engine/src/lib.rs

```diff
@@ -412,6 +412,15 @@ mod contract {
         sdk::return_output(&result.try_to_vec().sdk_expect("ERR_SERIALIZE"));
     }
 
+    #[no_mangle]
+    pub extern "C" fn get_block_hash() {
+        let block_height = sdk::read_input_borsh().sdk_unwrap();
```

Comment:
`get_block_hash_by_number` is implemented here (see the test in the PR for an example of how it is called). I do not think `get_current_block_hash` would be a very useful function since by the time you got the response back the result would no longer be "current".